server/pkcs11_test.go

```diff
@@ -0,0 +1,85 @@
+package server
+
+import (
+	"crypto/rand"
+	"io"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/cloudflare/gokeyless/internal/test/params"
+)
+
+var samples string = `pkcs11:id=0
+pkcs11:id=0;token=b
+pkcs11:id=b?pin-value=1234
+pkcs11:id=b?pin-value=1234&module-path=/a/b.so
+pkcs11:id=b;token=b?pin-value=1234&module-path=/a/b.so
+pkcs11:token=Thalesn6000;id=%00?module-path=/opt/nfast/toolkits/pkcs11/libcknfast.so&pin-value=1234
+pkcs11:token=SoftHSM2;id=%00;slot-id=1359138056?module-path=/usr/lib64/libsofthsm2.so&pin-value=1234
+pkcs11:token=YubiKey%20PIV;id=%00;slot-id=0?module-path=/usr/lib64/libykcs11.so&pin-value=123456
+pkcs11:token=Gemalto;id=%04;slot-id=0?module-path=/usr/lib/libCryptoki2_64.so&pin-value=abcd
+pkcs11:token=A;manufacturer=B;serial=C;model=D;library-manufacturer=E;library-description=F;library-version=G;object=H;type=I;id=J;slot-manufacturer=K;slot-description=L;slot-id=M?pin-source=N&pin-value=O&module-name=P&module-path=Q&max-sessions=R`
+
+var negatives string = `pkcs11:
+pkcs12:id=0
+pkcs11:id=b?pin-value=1234;module-path=/a/b.so`
+
+func TestRFC7512Parser(t *testing.T) {
+	tests := strings.Split(samples, "\n")
+	for _, uri := range tests {
+		if _, err := RFC7512Parser(uri); err != nil {
+			t.Fatal(err)
+		}
+	}
+	tests = strings.Split(negatives, "\n")
+	for _, uri := range tests {
+		if _, err := RFC7512Parser(uri); err == nil {
+			t.Fatalf("PKCS#11 Parser failed to detect a wrong URI: %s", uri)
+		}
+	}
+}
+
+func TestHSMSignConcurrencyRSASHA512(t *testing.T) {
```

Comment:
What is the purpose of this test? Are there others like it?

Reply:
This is similar to the benchmark function with "Concurrency" in its name, except the number of concurrent goroutines is fixed here.

Reply:
Rephrased: What are you testing? We already established that the concurrency benchmarks should be removed.

Reply:
I'll wait @cbroglie to explain this when he's back. He wanted these tests.

Reply:
I wanted these tests b/c I had concerns about crypto11's handling of lots of session handles, and the tests did identify some problems which we fixed.

But I had completely forgotten about the fact that gokeyless uses a limited pool of workers rather than each new connection getting its own goroutine, so my concerns about those crypto11 issues affecting gokeyless turned out to be unfounded. @Bren2010 is correct that the test can be removed.